packages/babel-helper-split-export-declaration/src/index.js

```diff
@@ -0,0 +1,65 @@
+import * as t from "@babel/types";
+
+const splitExportDeclaration = exportDeclaration => {
+  if (!exportDeclaration.isExportDeclaration()) {
+    throw new Error("Only export declarations can be splitted.");
+  }
+
+  // build specifiers that point back to this export declaration
+  const isDefault = exportDeclaration.isExportDefaultDeclaration();
+  const declaration = exportDeclaration.get("declaration");
+
+  if (isDefault) {
+    if (declaration.isIdentifier()) {
+      throw new Error("It doesn't make sense to split exported identifier.");
```

Comment:
Is this strictly necessary? It seems like we'd be able to change the logic for generating CmmonJS separately so

```
class Foo {}
export { Foo };
```

still transforms to something readable, even when it is split.

Reply:
What would be the value of splitting this? It would end up being something like
```js
class Foo {}
let Foo2 = Foo
export { Foo2 as Foo };
```

I've explicitly forbidden those cases as I'm throwing for them here, both for `ExportDefaultDeclaration` and `ExportNamedDeclaration`.

If you feel like those should be allowed, I'm really curious about when it could be useful.

Reply:
Yeah sorry, not sure what I was thinking when I wrote this. I do have a comment here, but I don't think this comment was actually it haha.

My question here is, what do we gain by throwing? There's nothing inherently wrong with splitting

```
export default foo;
```
into
```
var _default = foo;
export default _default;
```
and if the user doesn't want that they can already special-cause it before calling split.

Reply:
I don't think there is anything inherently wrong with this, I just don't see any gains in splitting this. I've added error to guard users against blindly using this helper and ending up with weirdo outputs - especially that helpers are mainly used as internal packages from what I understand.

I also don't actually feel about this really strongly, so if you feel that this error is unnecessary let me know and I'll remove it.

Reply:
I see the goal of splitting as "give me a standalone declaration that isn't tied to `export`. I'm curious what you see the goal as, if not that. Given that goal, whether the exported thing is an identifier or function/class declaration isn't really a special  case.

Reply:
I've focused more on the reasons why the splitting was already used - to simplify rest of the transform. Will change it to allow identifiers.